include/OpenSpaceToolkit/Astrodynamics/Estimator/ODLeastSquaresSolver.hpp

```diff
@@ -0,0 +1,152 @@
+#include <OpenSpaceToolkit/Core/Type/Size.hpp>
+
+#include <OpenSpaceToolkit/Mathematics/Object/Matrix.hpp>
+#include <OpenSpaceToolkit/Mathematics/Object/Vector.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/LocalOrbitalFrameFactory.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/NumericalSolver.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace estimator
+{
+
+using ostk::core::container::Array;
+using ostk::core::type::Real;
+using ostk::core::type::Shared;
+using ostk::core::type::Size;
+
+using ostk::mathematics::object::MatrixXd;
+using ostk::mathematics::object::VectorXd;
+
+using ostk::physics::coordinate::Frame;
+
+using ostk::astrodynamics::solver::LeastSquaresSolver;
+using ostk::astrodynamics::trajectory::LocalOrbitalFrameFactory;
+using ostk::astrodynamics::trajectory::NumericalSolver;
+using ostk::astrodynamics::trajectory::Orbit;
+using ostk::astrodynamics::trajectory::Propagator;
+using ostk::astrodynamics::trajectory::State;
+using ostk::astrodynamics::trajectory::state::CoordinateSubset;
+
+#define DEFAULT_ENVIRONMENT Environment::Default()                  // Default environment
+#define DEFAULT_NUMERICAL_SOLVER NumericalSolver::Default()         // Default numerical solver
+#define DEFAULT_LEAST_SQUARES_SOLVER LeastSquaresSolver::Default()  // Default least squares solver
+#define DEFAULT_ESTIMATION_FRAME Frame::GCRF()                      // Default estimation frame
+
+/// @brief Orbit Determination solver using least squares
+class ODLeastSquaresSolver
+{
+   public:
+    class Analysis
+    {
+       public:
+        /// @brief Constructor
+        Analysis(const State& aDeterminedState, const LeastSquaresSolver::Analysis& anAnalysis);
+
+        /// @brief Print analysis
+        friend std::ostream& operator<<(std::ostream& anOutputStream, const Analysis& anAnalysis);
+
+        /// @brief Print analysis
+        void print(std::ostream& anOutputStream) const;
+
+        State determinedState;
+        LeastSquaresSolver::Analysis solverAnalysis;
+    };
+
+    /// @brief Constructor
+    ///
+    /// @param anEnvironment Environment, Defaults to Environment::Default()
+    /// @param aNumericalSolver Numerical solver, Defaults to NumericalSolver::Default()
+    /// @param aSolver Least squares solver, Defaults to LeastSquaresSolver::Default()
+    ODLeastSquaresSolver(
+        const Environment& anEnvironment = DEFAULT_ENVIRONMENT,
+        const NumericalSolver& aNumericalSolver = DEFAULT_NUMERICAL_SOLVER,
+        const LeastSquaresSolver& aSolver = DEFAULT_LEAST_SQUARES_SOLVER,
+        const Shared<const Frame>& anEstimationFrameSPtr = DEFAULT_ESTIMATION_FRAME
+    );
```

Comment:
Do we really want this many defaults? constructing an ODLeastSquaresSolver() just like that feels a little weird...